orchestration/hca_orchestration/solids/create_snapshot.py

```diff
@@ -72,6 +85,45 @@ def get_completed_snapshot_info(context: AbstractComputeExecutionContext, job_id
     yield Output(snapshot_info_dict['id'])
 
 
+@solid(
+    config_schema=Field(Permissive({"validate_snapshot_name": Field(bool, default_value=True, is_required=False)})),
+    required_resource_keys={'data_repo_client', 'snapshot_config', 'hca_manage_config', 'data_repo_service'},
+)
+def get_snapshot_from_project(context: AbstractComputeExecutionContext) -> Any:
+    """
+    Use the snapshot_name to get the associated snapshot_id from TDR,
+    for the snapshot created in the previous pipeline step - cut_snapshot.
+    If there is no matching snapshot or dataset - fail
+    If the snapshot_name does not end in the release_tag fail - so that we know we are using the correct snapshot
+    """
+    data_repo_service: DataRepoService = context.resources.data_repo_service
+    dataset_name = context.resources.snapshot_config.dataset_name
+    snapshot_name = context.resources.snapshot_config.snapshot_name
+    release_tag = context.resources.snapshot_config.qualifier
+    dataset = data_repo_service.find_dataset(dataset_name)
+
+    # we need the dataset to get the billing profile id, which is needed to query (enumerate) the snapshot
+    if not dataset:
+        raise Failure(f"Snapshot not found for dataset name [dataset_name={dataset_name}]")
+    if not snapshot_name:
+        raise Failure(f"Snapshot name not found for snapshot name [snapshot_name={snapshot_name}]")
+    else:
+        if not release_tag:
+            raise Failure(f"Release tag not found for release tag [release_tag={release_tag}]. This is required.")
+        if not snapshot_name.endswith(release_tag):
+            raise Failure(f"Snapshot name does not end in current release tag [snapshot_name={snapshot_name}], \
+            [release_tag={release_tag}].")
+        else:
+            response = context.resources.data_repo_client.enumerate_snapshots(filter=dataset_name)
```

Comment:
I'm trying to remember the logic we discussed. The idea is to get the previously cut snapshot made for this particular release for the purposes of marking it public, correct? There may be further sanity checking we can do here. It seems like we want a precondition for marking the snapshot public to be that there is exactly one snapshot from the `enumerate_snapshots` query -- if there is more than one we are in uncharted territory and should throw up. Thoughts?

Reply:
Yes, that's the intent - make the snapshot from the previous step in the pipeline public, based on a project_id passed in via the partitions.
We can discuss this with Nate again, but we are less concerned with there being more than one (though that's good to check for too, as it's unexpected) but that this is the snapshot for this release.

IE I'm less concerned with two snapshot_name_080823_dcp30  snapshots going out than I am with snapshot_name_080823_dcp20 getting made public alongside snapshot_name_080823_dcp30

Especially because two snapshots with name snapshot_name_080823_dcp30 shouldn't be possible in TDR.